lib/sync-records.ts

```diff
@@ -51,25 +47,21 @@ export const syncWorkbookRecords = async ({
     };
   }
 
-  const validJobs = await validJobRecords(jobRecords);
-
-  console.log("Valid job records to sync", validJobs.length);
-
-  const upsertJobs = await upsertJobRecords(validJobs, {
-    userId,
-    organizationId,
-  });
-
-  const validEmployees = await validEmployeeRecords(employeeRecords);
+  // console.log("Valid job records to sync", numJobRecords);
 
-  console.log("Valid records to sync", validEmployees.length);
+  if (jobRecords) {
+    const upsertJobs = await upsertJobRecords(jobRecords, {
+      userId,
+      organizationId,
+    });
+  }
 
   // TODO: Refactor employee logic to its lib file
-  const validsManagersFirst = validEmployees.sort((a, b) => {
+  const validsManagersFirst = employeeRecords?.sort((a, b) => {
```

Comment:
What's the return type of employeeRecords, if it's an empty array we don't need this if check. Does this error without this if? An if with just a log isn't really worth it

Reply:
employeeRecords returns RecordsWithLinks | null | undefined

The await Promise.all(upsertEmployees) wont accept upsertEmployees as possibly undefined and yells with a overload error.

Putting the await Promise.all(upsertEmployees) in a conditional?

Getting this error

```
No overload matches this call.
  Overload 1 of 2, '(values: readonly unknown[] | []): Promise<unknown[] | []>', gave the following error.
    Argument of type 'Promise<void>[] | undefined' is not assignable to parameter of type 'readonly unknown[] | []'.
      Type 'undefined' is not assignable to type 'readonly unknown[] | []'.
  Overload 2 of 2, '(values: Iterable<void | PromiseLike<void>>): Promise<void[]>', gave the following error.
    Argument of type 'Promise<void>[] | undefined' is not assignable to parameter of type 'Iterable<void | PromiseLike<void>>'.
      Type 'undefined' is not assignable to type 'Iterable<void | PromiseLike<void>>'.ts(2769)
```

Reply:
if(employeeRecords) seems easiest